.github/workflows/run_smoke_test.yaml

```diff
@@ -0,0 +1,71 @@
+on: 
+  workflow_dispatch:
+  pull_request:
+    types: [opened, reopened]
+    branches:
+      - "main"
+  pull_request_review:
+      types: [submitted]  
+  push:
+    branches:
+      - "main"
+
+name: Smoke Test
+
+jobs:
+  Run-Smoke-Test:
+    runs-on: windows-latest
+    env:
+      SCUBA_GITHUB_AUTOMATION_CREDS: ${{ secrets.SCUBA_GITHUB_AUTOMATION_CREDS }}
+    defaults:
+          run:
+            shell: powershell 
+    permissions:
+      contents: read 
+    steps:
+    - name: Checkout repo code
+      uses: actions/checkout@v3
+    - name: Execute ScubaGear and Check Outputs
+      run: |
+        . Testing/Functional/SmokeTest/SmokeTestUtils.ps1
+        ./AllowBasicAuthentication.ps1 -RunAsAdmin
+
+        ##### Install all the dependencies
+        Install-SmokeTestExternalDependencies
+
+        # ScubaGear currently requires the provisioning of a certificate for using a ServicePrinicpal, rather than
+        # using Workload Identity Federation, which would ordinarily be preferred for calling Microsoft APIs from
+        # GitHub actions.
+        $AUTOMATION_CREDS = $env:SCUBA_GITHUB_AUTOMATION_CREDS | ConvertFrom-Json
+        $TestTenants = $AUTOMATION_CREDS.TestTenants
+        Write-Output "Identified $($TestTenants.Count) Test Tenants"
+
+        $TestContainers = @()
+        ForEach ($TestTenantObj in $TestTenants){
+          $Properties = Get-Member -InputObject $TestTenantObj -MemberType NoteProperty
+          $TestTenant = $TestTenantObj | Select-Object -ExpandProperty $Properties.Name
+          $OrgName = $TestTenant.DisplayName
+          $OrgName
+          $DomainName = $TestTenant.DomainName
+          $DomainName
+          $AppId = $TestTenant.AppId
+          $AppId
+          $PlainTextPassword = $TestTenant.CertificatePassword
+          $CertPwd = ConvertTo-SecureString -String $PlainTextPassword -Force -AsPlainText
+          $M365Env = $TestTenant.M365Env
+          $M365Env
+          try {
+            $Result = New-ServicePrincipalCertificate -EncodedCertificate $TestTenant.CertificateB64 -CertificatePassword $CertPwd
+            $Thumbprint = $Result[-1]
+          }
+          catch {
+            Write-Output "Failed to install certificate for $OrgName"
+          }
+
+          $TestContainers += New-PesterContainer -Path "Testing/Functional/SmokeTest/SmokeTest001.Tests.ps1" -Data @{ Thumbprint = $Thumbprint; Organization = $DomainName; AppId = $AppId; M365Environment = $M365Env }
```

Comment:
(style) I tend to prefer having a max line length.  Please consider limiting to 100c or other, even though its not currently enforced by the linter.

Reply:
I agree.  80 was old standard but to short now a days.

Reply:
Yes, I lived in the world of 80c for many years.  But that would be quite painful with PS (it certainly was for java too...) and isn't necessary with the monitors people use today.